### PR TITLE
xx-apt: add special case for ubuntu repo switch

### PR DESCRIPTION
**Pull request by tonistiigi**
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

File: base/xx-apt [hunk 1]
```diff
@@ -45,6 +45,50 @@ else
...
+  group1="amd64,i386"
+  group2="arm64,armhf,ppc64el,s390x"
+
+  nativearch="$(TARGETPLATFORM="" TARGETPAIR="" TARGETARCH="" TARGETOS="" xx-info arch)"
+
+  while IFS="" read -r p || [ -n "$p" ]; do
+
+    if echo "$p" | grep -E "^(deb|deb-src) "; then
+      case "${nativearch}" in
+        amd64 | 386)
```
Comment (tonistiigi) [inline base/xx-apt:63]: This fixes all active rules to the current arch group and adds a duplicate rule for the additional group.

Alternative would be to set current rules only to the current native arch and add detect the ubuntu distro and add only the known rules for that arch, each time fixed to individual arch.

Maybe the second one is better. wdyt? @crazy-max

**Comment by tonistiigi** [comment]
I removed 18.04 from CI. The llvm version is way too low for anything meaningful. `bionic-backports` might be enough but adding rules to them in cross-platform way is hacky as well. Other than clang suite, 18.04 seemed to pass.

File: base/xx-apt (hunk 1 above)
Comment (crazy-max) [reply base/xx-apt:63]: > Alternative would be to set current rules only to the current native arch and add detect the ubuntu distro and add only the known rules for that arch, each time fixed to individual arch.

Yes, the second solution with native arch might be better in the long run.

**Comment by crazy-max** [comment]
> I removed 18.04 from CI. The llvm version is way too low for anything meaningful. `bionic-backports` might be enough but adding rules to them in cross-platform way is hacky as well. Other than clang suite, 18.04 seemed to pass.

I've started the follow-up for #52 and on fedora 35 (stable) llvm version is quite new in my case 13.0.0~rc1.

**Review by crazy-max — APPROVED** [review]

File: base/xx-apt (hunk 1 above)
Comment (tonistiigi) [reply base/xx-apt:67]: Let's merge this one. I'll make a new PR and we can see if it actually improves.

**Pull request merged**